### PR TITLE
Spin up test VMs on GCE during a circle ci run and use them to run the integration tests.

### PR DESCRIPTION
**Pull request by tomwilkie**

### CODE REVIEW

File: test/gce/gce.sh [hunk 1]
```diff
@@ -0,0 +1,98 @@
+#!/bin/bash
+
+set -ex
+
+# Poor mans vagrant; cannot run vagrant on travis!
```
Comment (rade) [inline test/gce/gce.sh]: travis?

File: test/gce/gce.sh [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+echo deb https://get.docker.io/ubuntu docker main > /etc/apt/sources.list.d/docker.list;
+apt-get update -qq;
+apt-get install -q -y --force-yes --no-install-recommends lxc-docker ethtool;
+usermod -a -G docker vagrant;
+echo 'DOCKER_OPTS="-H unix:///var/run/docker.sock -H tcp://0.0.0.0:2375"' >> /etc/default/docker;
+service docker restart
+EOF
+		# It seems we need a short delay for docker to start up, so I put this in
+		# a separate ssh connection
+		ssh -t $name sudo docker run --rm -v /usr/local/bin:/target jpetazzo/nsenter
```
Comment (rade) [inline test/gce/gce.sh]: this is rather obscure and deserves an "install nsenter" comment.

File: test/gce/gce.sh [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+		gcloud compute instances delete --zone $ZONE -q $names
+	fi
+}
+
+function ExternalIPFor {
+	echo $(gcloud compute instances list $1 --format=yaml | grep "^    natIP\:" | cut -d: -f2)
+}
+
+function InternalIPFor {
+	echo $(gcloud compute instances list $1 --format=yaml | grep "^  networkIP\:" | cut -d: -f2)
```
Comment (rade) [inline test/gce/gce.sh]: why the sub-shells and echo?

**Comment by tomwilkie** [comment]
PTAL - in particular, where should gce.sh live (perhaps in bin/?)

**Comment by tomwilkie** [comment]
@paulbellamy

File: circle.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+test:
+  pre:
+    - ls -al /var/run/docker.sock
+    - mkdir -p $(dirname $SRCDIR)
+    - cp -r $(pwd)/ $SRCDIR
+  override:
+    - docker run -v /var/run/docker.sock:/run/docker.sock -v /home/ubuntu:/home/go weaveworks/weave-build
+    - cd $SRCDIR/test; ./gce/gce.sh setup
+    - cd $SRCDIR/test; . ./gce/gce.sh hosts; ./setup.sh
```
Comment (paulbellamy) [inline circle.yml]: Don't you mean: ```. <(./gce/gce.sh hosts);``` here, and the line below? As, I don't think ```.``` will take multiple args.

Comment (paulbellamy) [inline circle.yml]: Ah, not this does work too. TIL.

File: test/gce/gce.sh [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+		name="host$i"
+		gcloud compute instances create $name --image $IMAGE --zone $ZONE
+	done
+
+	gcloud compute config-ssh --ssh-key-file $SSH_KEY_FILE
+
+	hosts=
+	for i in $(seq 1 $NUM_HOSTS); do
+		name="host$i.$ZONE.$PROJECT"
+		ssh -t $name sudo bash -x -s <<EOF
```
Comment (paulbellamy) [inline test/gce/gce.sh]: Could extract this and the nsenter bit into a ```InstallDockerOn $name``` function, but not critical.

**Comment by paulbellamy** [comment]
Other than the above, LGTM.

**Pull request merged**